cryptol-saw-core/src/Verifier/SAW/Cryptol.hs

```diff
@@ -1807,56 +1850,38 @@ importMatches sc env (C.Let decl : matches) =
 
 
 --------------------------------------------------------------------------------
--- Utilities
-
-asCryptolTypeValue :: SC.TValue SC.Concrete -> Maybe (Either C.Kind C.Type)
-asCryptolTypeValue v =
-  case v of
-    SC.VBoolType -> return (Right C.tBit)
-    SC.VIntType -> return (Right C.tInteger)
-    SC.VIntModType n -> return (Right (C.tIntMod (C.tNum n)))
-    SC.VArrayType v1 v2 -> do
-      Right t1 <- asCryptolTypeValue v1
-      Right t2 <- asCryptolTypeValue v2
-      return (Right (C.tArray t1 t2))
-    SC.VVecType n v2 -> do
-      Right t2 <- asCryptolTypeValue v2
-      return (Right (C.tSeq (C.tNum n) t2))
-
-    SC.VDataType (primName -> "Prelude.Stream") [SC.TValue v1] [] ->
-        do Right t1 <- asCryptolTypeValue v1
-           return (Right (C.tSeq C.tInf t1))
-
-    SC.VDataType (primName -> "Cryptol.Num") [] [] ->
-      return (Left C.KNum)
-
-    SC.VDataType _ _ _ -> Nothing
-
-    SC.VUnitType -> return (Right (C.tTuple []))
-    SC.VPairType v1 v2 -> do
-      Right t1 <- asCryptolTypeValue v1
-      Right t2 <- asCryptolTypeValue v2
-      case C.tIsTuple t2 of
-        Just ts -> return (Right (C.tTuple (t1 : ts)))
-        Nothing -> return (Right (C.tTuple [t1, t2]))
-
-    SC.VPiType _nm v1 (SC.VNondependentPi v2) ->
-      do Right t1 <- asCryptolTypeValue v1
-         Right t2 <- asCryptolTypeValue v2
-         return (Right (C.tFun t1 t2))
-
-    SC.VSort s
-      | s == mkSort 0 -> return (Left C.KType)
-      | otherwise     -> Nothing
-
-    -- TODO?
-    SC.VPiType _nm _v1 (SC.VDependentPi _) -> Nothing
-    SC.VStringType -> Nothing
-    SC.VRecordType{} -> Nothing
-    SC.VRecursorType{} -> Nothing
-    SC.VTyTerm{} -> Nothing
-
-
+-- Utilities:
+
+-- | checkSAWCoreTypeChecks sc nm term mType - typeChecks (SAWCore) @term@.
+--     if @mType == Just type'@ then ensure the following
+--         term :: type'
+checkSAWCoreTypeChecks :: (Show i) =>
+  SharedContext -> i -> Term -> Maybe Term -> IO ()
```

Comment:
In my opinion, having the argument be a `Maybe Type` is unnecessarily complicated, as this function doesn't do anything interesting when the argument is a `Nothing` value. I'd prefer it if the type signature were instead:

```suggestion
  SharedContext -> i -> Term -> Term -> IO ()
```

Then, at the call site for `checkSAWCoreTypeChecks`, we can apply the function on the underlying value of a `Maybe Term` by doing something like:

```
traverse_ (checkSAWCoreTermTypeChecks sc ident) mTy
```

Which is a reasonably common idiom.